cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -203,23 +204,23 @@ public void onClose(KubernetesClientException e) {
     }
 
     /**
-     * Searches through the conditions in the supplied status instance and finds those whose type is
-     * {@link KafkaRebalanceStatus#REBALANCE_STATUS_CONDITION_TYPE}. If there are none it will return null. If there are
+     * Searches through the conditions in the supplied status instance and finds those whose type matches one of the values defined
+     * in the {@link KafkaRebalanceAssemblyOperator.State} enum. If there are none it will return null. If there are
      * more than one it will throw a RuntimeException. If there is only one it will return that Condition.
      *
      * @param status The KafkaRebalanceStatus instance whose conditions will be searched.
-     * @return The Condition instance from the supplied status that has type {@link KafkaRebalanceStatus#REBALANCE_STATUS_CONDITION_TYPE}.
-     *         If none are found then the method will return null.
-     * @throws RuntimeException If there is more than one Condition instance in the supplied status with the type
-     *                          {@link KafkaRebalanceStatus#REBALANCE_STATUS_CONDITION_TYPE}.
+     * @return The Condition instance from the supplied status that has a type value matching one of the values of the
+     *         {@link KafkaRebalanceAssemblyOperator.State} enum. If none are found then the method will return null.
+     * @throws RuntimeException If there is more than one Condition instance in the supplied status whose type matches one of the
+     *                          {@link KafkaRebalanceAssemblyOperator.State} enum values.
      */
     /* test */ protected Condition rebalanceStateCondition(KafkaRebalanceStatus status) {
         if (status.getConditions() != null) {
 
             List<Condition> statusConditions = status.getConditions()
                     .stream()
-                    .filter(condition -> condition.getType() != null &&
-                            condition.getType().equals(KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_TYPE))
+                    .filter(condition -> condition.getType() != null)
+                    .filter(condition -> Arrays.stream(State.values()).anyMatch(stateValue -> stateValue.toString().equals(condition.getType())))
```

Comment:
can we just do a `&&` adding the other condition in one `filter` operation instead of having two?